_posts/en/newsletters/2023-10-11-newsletter.md

```diff
@@ -0,0 +1,189 @@
+  discussed before (see [Newsletter #185][news185 txhash]) but this is
+  the first specification of the idea.  In addition to describing
+  exactly how the opcode would work, it also looks at mitigating some
+  potential downsides, such as full nodes potentially needing to hash up
+  to several megabytes of data every time the opcode is invoked.
+  Roose's draft includes a sample implementation of the opcode.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review
+Club][] meeting, highlighting some of the important questions and
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[util: Type-safe transaction identifiers][review club 28107] is a PR
+by Niklas Gögge (dergoegge) that improves type safety by introducing
+separate types for `txid` (the transaction identifier or hash that
+doesn't include the segwit witness data) and `wtxid` (same but includes
+the witness data), rather than both being represented by a `uint256`
+(a 256-bit integer, which can contain a SHA256 hash). This PR
+should have no operational effect; it aims to prevent future
+programming errors in which one kind of transaction ID is used where
+the other was intended. Such errors will be detected at compile time.
+
+To minimize disruption and ease review, these new types will initially
+be used in only one area of the code (the transaction "orphanage");
+future PRs will deploy the new types to other areas of the codebase.
+
+{% include functions/details-list.md
+  q0="What does it mean for a transaction identifier to be type-safe?
+      Why is that important or helpful? Are there any downsides?"
+  a0="A transaction identifier is considered type-safe when it undergoes
+      rigorous type-checking during compile-time, guaranteeing that
+      operations involving these identifiers are thoroughly validated
+      and constrained to their intended uses before the program is
+      executed, thus preventing type-related errors and ensuring
+      program reliability. No downsides were noted."
+  a0link="https://bitcoincore.reviews/28107#l-38"
+
+  q1="Rather than the new class types `Txid` and a `Wtxid` _inheriting_
+      from `uint256`, should they  _include_ (wrap) a `uint256`?
+      What are the tradeoffs?"
+  a1="Those classes could do that, but it would cause much more code
+      churn (many more source lines would need to be touched)."
+  a1link="https://bitcoincore.reviews/28107#l-43"
+
+  q2="Why is it better to enforce types at compile-time instead of at
+      run-time?"
+  a2="Developers discover errors quickly as they're coding, rather
+      than relying on writing extensive test suites to catch bugs in runtime
+      (and these tests may still miss some errors). However, testing would
+      still be useful since type safety won't prevent consistent use of the
+      wrong type of transaction ID (but consistently) in the first place."
```

Comment:
consistent (but consistently)
```suggestion
      still be useful since type safety won't prevent consistent use of the
      wrong type of transaction ID in the first place."
```